iocore/cache/test/main.cc

```diff
@@ -24,9 +24,23 @@
 #define CATCH_CONFIG_MAIN
 #include "main.h"
 
+#include <unistd.h>
+#include <filesystem>
+
 #define THREADS 1
 #define DIAGS_LOG_FILE "diags.log"
 
+// Create a new temp directory and return it
+std::string
+temp_prefix()
+{
+  char buffer[PATH_MAX];
+  snprintf(buffer, sizeof(buffer), "%s/cachetest.XXXXXX", getenv("TMPDIR") ?: "/tmp");
+  auto prefix = std::filesystem::path(mkdtemp(buffer));
```

Comment:
Unfortunately you need to use `ts::file` instead of `std::filesystem` because we weren't able to make it completely portable for all of our supported environments. `ts::file`  has a very similar API for this reason and hopefully, someday, we can KIWF and use the real thing.

We may need to add `create_directories` - I don't think that's there yet. We've been adding to `ts::file`  only as needed.

Reply:
Actually, I think what would be better is to put this in `ts::file` and have it return a `ts::file::path`.